modules/apps/frontend-data-set/frontend-data-set-test/src/testFunctional/macros/FrontendDataSet.macro

```diff
@@ -1,5 +1,58 @@
 definition {
 
+	macro chooseOptionNewFilterDropdown {
+		var key_option = ${key_option};
+
+		Type(
+			locator1 = "FrontendDataSetAdmin#NEW_FILTER_SELECT_OPTION",
+			value1 = ${key_option});
+	}
+
+	macro dateFilterSaveDates {
+		var key_dateNumber1 = ${key_dateNumber1};
+		var key_dateNumber2 = ${key_dateNumber2};
+
+		Click(
+			key_dateField = "From",
+			locator1 = "FrontendDataSetAdmin#DATEPICKER_BUTTON");
+
+		Click(
+			key_dateNumber = ${key_dateNumber1},
+			locator1 = "FrontendDataSetAdmin#CHOOSE_DATE_DATEPICKER_MENU");
+
+		Click(
+			key_dateField = "From",
+			locator1 = "FrontendDataSetAdmin#DATEPICKER_BUTTON");
+
+		Click(
+			key_dateField = "To",
+			locator1 = "FrontendDataSetAdmin#DATEPICKER_BUTTON");
+
+		Click(
+			key_dateNumber = ${key_dateNumber2},
+			locator1 = "FrontendDataSetAdmin#CHOOSE_DATE_DATEPICKER_MENU");
+
+		Click(
+			key_dateField = "To",
+			locator1 = "FrontendDataSetAdmin#DATEPICKER_BUTTON");
+
+		Click(locator1 = "FrontendDataSetAdmin#NEW_FILTER_SAVE_BUTTON");
+	}
+
+	macro selectFromFieldDates {
+		Click(
+			key_dateField = "From",
+			locator1 = "FrontendDataSetAdmin#DATEPICKER_BUTTON");
+
+		Click(
+			locator1 = "FrontendDataSetAdmin#CHOOSE_DATE_DATEPICKER_MENU",
+			value1 = ${key_dateNumber});
+
+		Click(
+			key_dateField = "From",
+			locator1 = "FrontendDataSetAdmin#DATEPICKER_BUTTON");
+	}
```

Comment:
Is it possible to use YUICalendarWidget.addDate() macro instead of creating a new one?